tcp_endpoint/src/tcp_endpoint/RosUnityHandshakeService.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+
+import rospy
+from tcp_endpoint.srv import RosUnityHandshake, RosUnityHandshakeResponse
+
+class RosUnityHandshakeService:
+    """
+    Class to auto-detect Unity IP.
+    """
+    def __init__(self, tcp_sender):
+        """
+        Args:
+            tcp_sender:     sends messages to Unity
+        """
+        self.srv_class = RosUnityHandshake._request_class()
+        self.tcp_sender = tcp_sender
+        self.incoming_ip = ''
+
+    def set_thread(self, tcp_client_thread):
+        self.incoming_ip = tcp_client_thread.incoming_ip
+
+    def send(self, data):

```

Comment:
Is this solution able to support multiple Unity instances? An example use case might be, I made some recent changes to my ML model that lives as a ROS service and want to test it across a few different scenarios as part of my CI tests.

Reply:
Right now a single tcp_server can receive from, but can't send to, multiple simultaneous Unity instances. However, it will seamlessly switch whenever a new instance sends it a handshake, so you should be able to test it across a few different scenarios, one at a time, without restarting your ROS component.

If you need them to connect simultaneously, I imagine it should be possible to start multiple tcp_servers that listen on different ports?